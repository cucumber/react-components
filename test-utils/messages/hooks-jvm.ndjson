{"meta":{"protocolVersion":"13.0.1","implementation":{"name":"cucumber-jvm","version":"6.6.1-SNAPSHOT"},"runtime":{"name":"OpenJDK 64-Bit Server VM","version":"11.0.8+11"},"os":{"name":"Mac OS X"},"cpu":{"name":"x86_64"}}}
{"testRunStarted":{"timestamp":{"seconds":"1599042650","nanos":118359000}}}
{"source":{"uri":"file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature","data":"Feature: Hooks\n  Hooks are special steps that run before or after each scenario\u0027s steps.\n  They can also conditionally target specific scenarios, using tag expressions\n\n  Scenario: no tags, passed step\n    When a step passes\n\n  Scenario: no tags, failed step\n    When a step throws an exception\n\n  Scenario: no tags, undefined step\n    When a step throws an exception\n\n  @some-tag\n  Scenario: with a tag, passed step\n    When a step passes\n","mediaType":"text/x.cucumber.gherkin+plain"}}
{"gherkinDocument":{"uri":"file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature","feature":{"location":{"line":1,"column":1},"language":"en","keyword":"Feature","name":"Hooks","description":"  Hooks are special steps that run before or after each scenario\u0027s steps.\n  They can also conditionally target specific scenarios, using tag expressions","children":[{"scenario":{"location":{"line":5,"column":3},"keyword":"Scenario","name":"no tags, passed step","steps":[{"location":{"line":6,"column":5},"keyword":"When ","text":"a step passes","id":"550cf1b7-9bd2-426f-a66b-dce11c977f10"}],"id":"830608d3-bcc3-440d-aa3c-613311355cab"}},{"scenario":{"location":{"line":8,"column":3},"keyword":"Scenario","name":"no tags, failed step","steps":[{"location":{"line":9,"column":5},"keyword":"When ","text":"a step throws an exception","id":"1eb3b3cf-d754-43ac-9a4e-04f5d11b0557"}],"id":"c751995c-ed16-43fb-a399-ddf6087e444b"}},{"scenario":{"location":{"line":11,"column":3},"keyword":"Scenario","name":"no tags, undefined step","steps":[{"location":{"line":12,"column":5},"keyword":"When ","text":"a step throws an exception","id":"f3430236-60e9-4275-ab71-3ea49a74fb4a"}],"id":"3b623a81-4e95-4d93-968d-a7e6da4fe456"}},{"scenario":{"location":{"line":15,"column":3},"tags":[{"location":{"line":14,"column":3},"name":"@some-tag","id":"a1e3ece6-3dfd-4b0c-b75e-7b72e21e356b"}],"keyword":"Scenario","name":"with a tag, passed step","steps":[{"location":{"line":16,"column":5},"keyword":"When ","text":"a step passes","id":"788bf903-d3ad-4240-a09d-d377943a0e18"}],"id":"2b2aa9aa-3e58-4b24-8d73-45a50636cd69"}}]}}}
{"pickle":{"id":"3d701a7e-e06d-45a9-be1f-b0c6a0f8ca1f","uri":"file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature","name":"no tags, passed step","language":"en","steps":[{"text":"a step passes","id":"f173231a-0b45-4962-acab-f7e5fef14641","astNodeIds":["550cf1b7-9bd2-426f-a66b-dce11c977f10"]}],"astNodeIds":["830608d3-bcc3-440d-aa3c-613311355cab"]}}
{"pickle":{"id":"36b608b6-0d49-42c2-827b-e0303b0e6cd8","uri":"file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature","name":"no tags, failed step","language":"en","steps":[{"text":"a step throws an exception","id":"fc28ebee-01de-43f6-aab6-390365d96e16","astNodeIds":["1eb3b3cf-d754-43ac-9a4e-04f5d11b0557"]}],"astNodeIds":["c751995c-ed16-43fb-a399-ddf6087e444b"]}}
{"pickle":{"id":"b1b23641-8dad-4146-95c3-e37569dcf3a5","uri":"file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature","name":"no tags, undefined step","language":"en","steps":[{"text":"a step throws an exception","id":"b3a0786a-4242-41b7-9352-f4c13bc1addb","astNodeIds":["f3430236-60e9-4275-ab71-3ea49a74fb4a"]}],"astNodeIds":["3b623a81-4e95-4d93-968d-a7e6da4fe456"]}}
{"pickle":{"id":"368db5cf-be44-4516-b2a1-29ad1092c7c6","uri":"file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature","name":"with a tag, passed step","language":"en","steps":[{"text":"a step passes","id":"5bc5974a-fefe-46ca-b88b-8885dee322aa","astNodeIds":["788bf903-d3ad-4240-a09d-d377943a0e18"]}],"tags":[{"name":"@some-tag","astNodeId":"a1e3ece6-3dfd-4b0c-b75e-7b72e21e356b"}],"astNodeIds":["2b2aa9aa-3e58-4b24-8d73-45a50636cd69"]}}
{"hook":{"id":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"before"}}}}
{"stepDefinition":{"id":"b4845687-276c-49cd-a25e-656712b55989","pattern":{"source":"a step passes"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"aStepPasses"}}}}
{"stepDefinition":{"id":"12afaea2-c093-417b-9f50-7d33ca9373d7","pattern":{"source":"a step throws an exception"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"test"}}}}
{"hook":{"id":"1a24148a-7abd-4587-8380-115a722fed84","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"afterWithException"}}}}
{"hook":{"id":"997b8d30-25fc-4257-a08b-f54e0575467c","tagExpression":"@some-tag or @some-other-tag","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"taggedAfterWithException"}}}}
{"testCase":{"id":"f9b16c4d-9c4c-416c-bd9d-f4c505181e7d","pickleId":"3d701a7e-e06d-45a9-be1f-b0c6a0f8ca1f","testSteps":[{"id":"915b269a-fc4c-443f-bfb7-52d697a7c4e1","hookId":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a"},{"id":"54635301-9579-48ff-ab70-a9017c794988","pickleStepId":"f173231a-0b45-4962-acab-f7e5fef14641","stepDefinitionIds":["b4845687-276c-49cd-a25e-656712b55989"],"stepMatchArgumentsLists":[{}]},{"id":"08291ef5-9624-4c43-a395-1d0909f90013","hookId":"1a24148a-7abd-4587-8380-115a722fed84"}]}}
{"testCaseStarted":{"timestamp":{"seconds":"1599042650","nanos":130931000},"testCaseId":"f9b16c4d-9c4c-416c-bd9d-f4c505181e7d","id":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":132478000},"testStepId":"915b269a-fc4c-443f-bfb7-52d697a7c4e1","testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":125000}},"timestamp":{"seconds":"1599042650","nanos":132603000},"testStepId":"915b269a-fc4c-443f-bfb7-52d697a7c4e1","testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":132793000},"testStepId":"54635301-9579-48ff-ab70-a9017c794988","testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":77000}},"timestamp":{"seconds":"1599042650","nanos":132870000},"testStepId":"54635301-9579-48ff-ab70-a9017c794988","testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":133093000},"testStepId":"08291ef5-9624-4c43-a395-1d0909f90013","testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Exception in hook\n\tat io.cucumber.compatibility.hooks.Hooks.afterWithException(Hooks.java:24)\n","duration":{"nanos":375000}},"timestamp":{"seconds":"1599042650","nanos":133468000},"testStepId":"08291ef5-9624-4c43-a395-1d0909f90013","testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"testCaseFinished":{"timestamp":{"seconds":"1599042650","nanos":133835000},"testCaseStartedId":"48a72ce0-3448-4225-81b6-fb3b04416fe0"}}
{"hook":{"id":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"before"}}}}
{"stepDefinition":{"id":"4c2060a2-ef72-46cf-aee2-656bc8b7edc1","pattern":{"source":"a step passes"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"aStepPasses"}}}}
{"stepDefinition":{"id":"33758624-2727-4044-b014-f76a61a43578","pattern":{"source":"a step throws an exception"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"test"}}}}
{"hook":{"id":"1a24148a-7abd-4587-8380-115a722fed84","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"afterWithException"}}}}
{"hook":{"id":"997b8d30-25fc-4257-a08b-f54e0575467c","tagExpression":"@some-tag or @some-other-tag","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"taggedAfterWithException"}}}}
{"testCase":{"id":"0b68abc4-2204-4e7e-a5c5-d8975ebb6211","pickleId":"36b608b6-0d49-42c2-827b-e0303b0e6cd8","testSteps":[{"id":"290e6639-d73f-4735-8114-171b413f8167","hookId":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a"},{"id":"6a6bbbfa-9ce7-4b08-abb1-6cd350b191a9","pickleStepId":"fc28ebee-01de-43f6-aab6-390365d96e16","stepDefinitionIds":["33758624-2727-4044-b014-f76a61a43578"],"stepMatchArgumentsLists":[{}]},{"id":"00a2e5f1-64fe-4e6c-b0dd-fcff1b303547","hookId":"1a24148a-7abd-4587-8380-115a722fed84"}]}}
{"testCaseStarted":{"timestamp":{"seconds":"1599042650","nanos":135247000},"testCaseId":"0b68abc4-2204-4e7e-a5c5-d8975ebb6211","id":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":135526000},"testStepId":"290e6639-d73f-4735-8114-171b413f8167","testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":208000}},"timestamp":{"seconds":"1599042650","nanos":135734000},"testStepId":"290e6639-d73f-4735-8114-171b413f8167","testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":135958000},"testStepId":"6a6bbbfa-9ce7-4b08-abb1-6cd350b191a9","testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Boom\n\tat io.cucumber.compatibility.hooks.Hooks.test(Hooks.java:19)\n\tat ✽.a step throws an exception(file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature:9)\n","duration":{"nanos":399000}},"timestamp":{"seconds":"1599042650","nanos":136357000},"testStepId":"6a6bbbfa-9ce7-4b08-abb1-6cd350b191a9","testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":136899000},"testStepId":"00a2e5f1-64fe-4e6c-b0dd-fcff1b303547","testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Exception in hook\n\tat io.cucumber.compatibility.hooks.Hooks.afterWithException(Hooks.java:24)\n","duration":{"nanos":327000}},"timestamp":{"seconds":"1599042650","nanos":137226000},"testStepId":"00a2e5f1-64fe-4e6c-b0dd-fcff1b303547","testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"testCaseFinished":{"timestamp":{"seconds":"1599042650","nanos":137520000},"testCaseStartedId":"e47dc8fd-4ba8-413a-b358-f481e51051e1"}}
{"hook":{"id":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"before"}}}}
{"stepDefinition":{"id":"850eb4b0-613d-4a91-9247-1bb38ffd99ed","pattern":{"source":"a step passes"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"aStepPasses"}}}}
{"stepDefinition":{"id":"a7dd8dc0-f6f5-4f24-8d5f-6cf5bc96ff48","pattern":{"source":"a step throws an exception"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"test"}}}}
{"hook":{"id":"1a24148a-7abd-4587-8380-115a722fed84","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"afterWithException"}}}}
{"hook":{"id":"997b8d30-25fc-4257-a08b-f54e0575467c","tagExpression":"@some-tag or @some-other-tag","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"taggedAfterWithException"}}}}
{"testCase":{"id":"e17b5c99-f026-46b9-abfc-38b58198b9ea","pickleId":"b1b23641-8dad-4146-95c3-e37569dcf3a5","testSteps":[{"id":"f3cf363e-85fb-47a2-bf83-903b4e9d9d20","hookId":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a"},{"id":"af1245e1-7086-43ee-a824-921db38d05af","pickleStepId":"b3a0786a-4242-41b7-9352-f4c13bc1addb","stepDefinitionIds":["a7dd8dc0-f6f5-4f24-8d5f-6cf5bc96ff48"],"stepMatchArgumentsLists":[{}]},{"id":"22708cec-7127-493b-94cb-6165f03f8924","hookId":"1a24148a-7abd-4587-8380-115a722fed84"}]}}
{"testCaseStarted":{"timestamp":{"seconds":"1599042650","nanos":139825000},"testCaseId":"e17b5c99-f026-46b9-abfc-38b58198b9ea","id":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":140835000},"testStepId":"f3cf363e-85fb-47a2-bf83-903b4e9d9d20","testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":404000}},"timestamp":{"seconds":"1599042650","nanos":141239000},"testStepId":"f3cf363e-85fb-47a2-bf83-903b4e9d9d20","testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":141580000},"testStepId":"af1245e1-7086-43ee-a824-921db38d05af","testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Boom\n\tat io.cucumber.compatibility.hooks.Hooks.test(Hooks.java:19)\n\tat ✽.a step throws an exception(file:///Users/Aslak.Hellesoy/git/cucumber/cucumber-jvm/compatibility/src/test/resources/features/hooks/hooks.feature:12)\n","duration":{"nanos":981000}},"timestamp":{"seconds":"1599042650","nanos":142561000},"testStepId":"af1245e1-7086-43ee-a824-921db38d05af","testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":143216000},"testStepId":"22708cec-7127-493b-94cb-6165f03f8924","testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Exception in hook\n\tat io.cucumber.compatibility.hooks.Hooks.afterWithException(Hooks.java:24)\n","duration":{"nanos":313000}},"timestamp":{"seconds":"1599042650","nanos":143529000},"testStepId":"22708cec-7127-493b-94cb-6165f03f8924","testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"testCaseFinished":{"timestamp":{"seconds":"1599042650","nanos":143809000},"testCaseStartedId":"20d9009c-44d8-42e0-9d22-902963d341d8"}}
{"hook":{"id":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"before"}}}}
{"stepDefinition":{"id":"f9094414-cfe2-44ac-90d4-3a5b95b2e974","pattern":{"source":"a step passes"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"aStepPasses"}}}}
{"stepDefinition":{"id":"1eb78876-7788-4ce0-8ac2-a935d60ec386","pattern":{"source":"a step throws an exception"},"sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"test"}}}}
{"hook":{"id":"1a24148a-7abd-4587-8380-115a722fed84","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"afterWithException"}}}}
{"hook":{"id":"997b8d30-25fc-4257-a08b-f54e0575467c","tagExpression":"@some-tag or @some-other-tag","sourceReference":{"javaMethod":{"className":"io.cucumber.compatibility.hooks.Hooks","methodName":"taggedAfterWithException"}}}}
{"testCase":{"id":"bf1c0c7e-95d0-4fa5-a04d-41bc9f0232db","pickleId":"368db5cf-be44-4516-b2a1-29ad1092c7c6","testSteps":[{"id":"5d5decff-b9f1-4df6-ad28-c9f51b23b534","hookId":"a5dd5af8-b696-4f82-a9d3-347741ed7d8a"},{"id":"e23845a3-42f4-4c72-b943-20dfeaa8332d","pickleStepId":"5bc5974a-fefe-46ca-b88b-8885dee322aa","stepDefinitionIds":["f9094414-cfe2-44ac-90d4-3a5b95b2e974"],"stepMatchArgumentsLists":[{}]},{"id":"a1478b5d-24d5-422d-b726-e3c5880bde7b","hookId":"997b8d30-25fc-4257-a08b-f54e0575467c"},{"id":"a1dde710-3801-4fcb-a646-ec868dbc0825","hookId":"1a24148a-7abd-4587-8380-115a722fed84"}]}}
{"testCaseStarted":{"timestamp":{"seconds":"1599042650","nanos":145138000},"testCaseId":"bf1c0c7e-95d0-4fa5-a04d-41bc9f0232db","id":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":145408000},"testStepId":"5d5decff-b9f1-4df6-ad28-c9f51b23b534","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":95000}},"timestamp":{"seconds":"1599042650","nanos":145503000},"testStepId":"5d5decff-b9f1-4df6-ad28-c9f51b23b534","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":145601000},"testStepId":"e23845a3-42f4-4c72-b943-20dfeaa8332d","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":95000}},"timestamp":{"seconds":"1599042650","nanos":145696000},"testStepId":"e23845a3-42f4-4c72-b943-20dfeaa8332d","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":145812000},"testStepId":"a1478b5d-24d5-422d-b726-e3c5880bde7b","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Exception in conditional hook\n\tat io.cucumber.compatibility.hooks.Hooks.taggedAfterWithException(Hooks.java:29)\n","duration":{"nanos":311000}},"timestamp":{"seconds":"1599042650","nanos":146123000},"testStepId":"a1478b5d-24d5-422d-b726-e3c5880bde7b","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepStarted":{"timestamp":{"seconds":"1599042650","nanos":146639000},"testStepId":"a1dde710-3801-4fcb-a646-ec868dbc0825","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testStepFinished":{"testStepResult":{"status":"FAILED","message":"java.lang.Exception: Exception in hook\n\tat io.cucumber.compatibility.hooks.Hooks.afterWithException(Hooks.java:24)\n","duration":{"nanos":306000}},"timestamp":{"seconds":"1599042650","nanos":146945000},"testStepId":"a1dde710-3801-4fcb-a646-ec868dbc0825","testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testCaseFinished":{"timestamp":{"seconds":"1599042650","nanos":147302000},"testCaseStartedId":"bcaa81d4-691e-4192-a535-34ebc6baa98b"}}
{"testRunFinished":{"timestamp":{"seconds":"1599042650","nanos":147456000}}}
