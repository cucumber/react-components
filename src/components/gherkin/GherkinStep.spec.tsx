import { Query as GherkinQuery } from '@cucumber/gherkin-utils'
import * as messages from '@cucumber/messages'
import { Query as CucumberQuery } from '@cucumber/query'
import React from 'react'

import { render } from '../../../test-utils/index.js'
import { GherkinStep } from './GherkinStep.js'

describe('<GherkinStep>', () => {
  it('renders', () => {
    const step: messages.Step = {
      keyword: 'Given',
      text: 'the 48 pixies',
      location: { column: 1, line: 1 },
      id: '123',
    }

    class StubCucumberQuery extends CucumberQuery {
      public getStepMatchArgumentsLists(): messages.StepMatchArgumentsList[] {
        return [
          {
            stepMatchArguments: [
              {
                group: {
                  start: 4,
                  value: '48',
                  children: [],
                },
              },
            ],
          },
        ]
      }
    }

    class StubGherkinQuery extends GherkinQuery {
      getPickleStepIds(): string[] {
        return ['dummy-id']
      }
    }

    const { asFragment } = render(<GherkinStep step={step} hasExamples={false} />, {
      gherkinQuery: new StubGherkinQuery(),
      cucumberQuery: new StubCucumberQuery(),
    })

    expect(asFragment()).toMatchSnapshot()
  })
})
