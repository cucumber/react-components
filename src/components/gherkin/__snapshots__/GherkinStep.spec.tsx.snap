// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GherkinStep> renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span
      class="status"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-question fa-1x "
        data-icon="circle-question"
        data-prefix="fas"
        data-status="UNKNOWN"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24V250.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1H222.6c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM288 352c0 17.7-14.3 32-32 32s-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="content"
    >
      <div>
        <a
          class="anchor"
          href="#123"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-link "
            data-icon="link"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M579.8 267.7c56.5-56.5 56.5-148 0-204.5c-50-50-128.8-56.5-186.3-15.4l-1.6 1.1c-14.4 10.3-17.7 30.3-7.4 44.6s30.3 17.7 44.6 7.4l1.6-1.1c32.1-22.9 76-19.3 103.8 8.6c31.5 31.5 31.5 82.5 0 114L422.3 334.8c-31.5 31.5-82.5 31.5-114 0c-27.9-27.9-31.5-71.8-8.6-103.8l1.1-1.6c10.3-14.4 6.9-34.4-7.4-44.6s-34.4-6.9-44.6 7.4l-1.1 1.6C206.5 251.2 213 330 263 380c56.5 56.5 148 56.5 204.5 0L579.8 267.7zM60.2 244.3c-56.5 56.5-56.5 148 0 204.5c50 50 128.8 56.5 186.3 15.4l1.6-1.1c14.4-10.3 17.7-30.3 7.4-44.6s-30.3-17.7-44.6-7.4l-1.6 1.1c-32.1 22.9-76 19.3-103.8-8.6C74 372 74 321 105.5 289.5L217.7 177.2c31.5-31.5 82.5-31.5 114 0c27.9 27.9 31.5 71.8 8.6 103.9l-1.1 1.6c-10.3 14.4-6.9 34.4 7.4 44.6s34.4 6.9 44.6-7.4l1.1-1.6C433.5 260.8 427 182 377 132c-56.5-56.5-148-56.5-204.5 0L60.2 244.3z"
              fill="currentColor"
            />
          </svg>
        </a>
        <h3
          id="123"
        >
          <span>
            Given
          </span>
          <span>
            <span
              class="highlight"
            >
              the 
            </span>
            <span
              aria-label="48"
              title="anonymous"
            >
              <span
                class="highlight"
              >
                48
              </span>
            </span>
            <span
              class="highlight"
            >
               pixies
            </span>
          </span>
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
